File: frontend/angular-secretroom-website/src/app/model/order-review.ts
import {Address} from "./address";
import {OrderItem} from "./order";
import {ShippingOption} from "./shipping-option";
import {Customer} from "./customer";

export interface OrderReview {
  orderNumber: string;
  shippingAddress: Address;
  items: OrderItem[];
  subtotal: number;
  totalAmountOrder: number;
  shippingOption: ShippingOption;
  customer: Customer;

}

File: backend/src/main/java/com/secretroomwebsite/order/OrderReview.java
package com.secretroomwebsite.order;

import com.secretroomwebsite.adress.Address;
import com.secretroomwebsite.customer.Customer;
import com.secretroomwebsite.shipping.Shipping;

import java.time.LocalDate;
import java.util.Date;
import java.util.List;

public record OrderReview(
        String orderNumber,

        Date orderDate,
        Address shippingAddress,
        List<OrderItem> items,
        Double totalAmountOrder,
        Shipping shippingOption,
        Double subtotal,
        Customer customer
) { }